network-api/networkapi/tests.py

```diff
@@ -465,3 +467,136 @@ def test_software_page_export_fields(self):
+            # page fields
+            title='Software Percy Product',
+            first_published_at=datetime(2025, 1, 1, tzinfo=timezone.utc),
+            last_published_at=datetime(2025, 1, 1, tzinfo=timezone.utc),
+            parent=pni_homepage,
+            cloudinary_image=cloudinary_image,
+            # product fields
+            privacy_ding=True,
+            adult_content=True,
+            uses_wifi=True,
+            uses_bluetooth=True,
+            review_date=date(2025, 1, 1),
+            company='Percy Corp',
+            blurb='This is a general product specifically created for visual regression testing',
+            product_url='http://example.com/general-percy',
+            price=999,
+            worst_case='Visual regression fails',
+            # software product fields
+            handles_recordings_how='Sample software recording description',
+            recording_alert='Yes',
+        )
+
+    def test_fields_remain_untouched_and_wagtail_image_is_changed(self):
+        # Test fields before the command is run
+        # Given a page has a cloudinary image,
+        # When converting cloudinary images to wagtail images,
+        # Then wagtail images should be the same image as the old cloudinary image
+        # Ensure they have cloudinary fields
+        software_product_page = Page.objects.get(pk=self.software_product_page.id).specific
+        general_product_page = Page.objects.get(pk=self.general_product_page.id).specific
+        # Assert there are cloudinary images that will be converted to Wagtail images.
+        self.assertTrue(hasattr(software_product_page, 'cloudinary_image'))
+        self.assertTrue(hasattr(general_product_page, 'cloudinary_image'))
+        # Assert General Product Page fields are what we expect when setting them in the setUp() method
+        self.assertEqual(general_product_page.offline_capable, 'Yes')
+        self.assertEqual(
+            general_product_page.offline_use_description,
+            'Although it is unclear how offline capabilities work'
+        )
+        # Assert Software Product Page fields are what we expect when setting them in the setUp() method
+        self.assertEqual(software_product_page.recording_alert, 'Yes')
+        self.assertEqual(software_product_page.handles_recordings_how, 'Sample software recording description')
+        # Assert the image field and cloudinary fields are not the same
+        self.assertNotEqual(
+            software_product_page.image.filename,
+            f'{software_product_page.cloudinary_image.public_id}.{software_product_page.cloudinary_image.format}',
+        )
+        self.assertNotEqual(
+            general_product_page.image.filename,
+            f'{general_product_page.cloudinary_image.public_id}.{general_product_page.cloudinary_image.format}',
+        )
+
+        # Run the management command which will convert Cloudinary images to Wagtail images.
+        call_command("migrate_cloudinary")
```

Comment:
before we run this, let's convert `software_product_page` and `general_product_page` to dicts so we can use those as references later